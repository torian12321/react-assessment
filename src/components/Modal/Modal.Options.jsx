import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Options = styled.div`
  display: flex;
  justify-content: flex-end;
  gap: 12px;
  padding-top: 40px;
`;

export const ModalOptions = ({ children }) => (children ? (
  <Options>
    {children}
  </Options>
) : null);

ModalOptions.propTypes = {
  children: PropTypes.node,
};
ModalOptions.defaultProps = {
  children: null,
};
