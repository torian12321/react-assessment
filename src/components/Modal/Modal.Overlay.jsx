import styled from 'styled-components';

export const ModalOverlay = styled.div`
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background: black;
  opacity: 0.8;
  z-index: 1000;
`;
