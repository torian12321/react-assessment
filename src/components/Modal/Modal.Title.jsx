import React from 'react';
import styled from 'styled-components';

const Title = styled.div`
  color: black;
  margin-bottom: 12px;
  font-size: 16px;
  border-bottom: 1px solid #aaa;
`;

export const ModalTitle = ({ title }) => (title ? (
  <Title>
    {title}
  </Title>
) : null);
