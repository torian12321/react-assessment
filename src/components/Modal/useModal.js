import { useState, useEffect } from 'react';
import { useKeyPress } from '@hooks/useKeyPress';

/**
 * Hook returning <Modal /> controllers
 *
 */
const KEY_ESC = 27;

const useModal = () => {
  const [isVisible, setIsVisible] = useState(false);
  const escIsPressed = useKeyPress(KEY_ESC);
  const hiddeModal = () => setIsVisible(false);
  const showModal = () => setIsVisible(true);

  useEffect(() => {
    // Remove the Modal from the screen when ESC is pressed
    if (isVisible && escIsPressed) {
      hiddeModal();
    }
  }, [isVisible, escIsPressed]);

  return [
    {
      show: showModal,
      hidde: hiddeModal,
    },
    isVisible,
  ];
};

export { useModal };
