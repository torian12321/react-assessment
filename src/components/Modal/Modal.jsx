import React from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';
import { ModalOverlay } from './Modal.Overlay';
import { ModalTitle } from './Modal.Title';
import { ModalOptions } from './Modal.Options';

const ModalWrapper = styled.div`
  position: fixed;
  top: 20px;
  left: 50%;
  transform: translateX(-50%);

  width: 80%;
  max-width: 600px;  
  padding: 20px 16px;
  background: white;
  border: 1px solid red;
  border-radius: 8px;
  z-index: 1001;
`;

export const Modal = ({ isVisible = false, children, title }) => {
  const { body: _dBody } = document;

  return (isVisible && children) ? ReactDOM.createPortal(
    <>
      <ModalOverlay />
      <ModalWrapper>
        <ModalTitle title={title} />
        {children}
      </ModalWrapper>
    </>,
    _dBody
  ) : null;
};

Modal.Options = ModalOptions;
