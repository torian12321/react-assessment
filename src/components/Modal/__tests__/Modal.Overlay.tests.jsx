import React from 'react';
import { render } from '@testing-library/react';
import { ModalOverlay } from '../Modal.Overlay';
import 'jest-styled-components';

it('renders correctly', () => {
  const { container } = render(<ModalOverlay />);

  expect(container).toMatchSnapshot();
});
