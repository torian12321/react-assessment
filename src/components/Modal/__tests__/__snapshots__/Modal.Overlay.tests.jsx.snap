// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background: black;
  opacity: 0.8;
  -webkit-animation: laHpMQ 0.5s ease-in 1;
  animation: laHpMQ 0.5s ease-in 1;
}

<div>
  <div
    class="c0"
  />
</div>
`;
